mqtt/init.lua

```diff
@@ -81,6 +81,183 @@ function mqtt.run_sync(cl)
 	end
 end
 
+
+--- Validates a topic with wildcards.
+-- @param t (string) wildcard topic to validate
+-- @return topic, or false+error
+function mqtt.validate_subscribe_topic(t)
+	if type(t) ~= "string" then
+		return false, "not a string"
+	end
+	if #t < 1 then
+		return false, "minimum topic length is 1"
+	end
+	do
+		local _, count = t:gsub("#", "")
+		if count > 1 then
+			return false, "wildcard '#' may only appear once"
+		end
+		if count == 1 then
+			if t ~= "#" and not t:find("/#$") then
+				return false, "wildcard '#' must be the last character, and be prefixed with '/' (unless the topic is '#')"
+			end
+		end
+	end
+	do
+		local t1 = "/"..t.."/"
+		local i = 1
+		while i do
+			i = t1:find("+", i)
+			if i then
+				if t1:sub(i-1, i+1) ~= "/+/" then
+					return false, "wildcard '+' must be enclosed between '/' (except at start/end)"
+				end
+				i = i + 1
+			end
+		end
+	end
+	return t
+end
+
+--- Validates a topic without wildcards.
+-- @param t (string) topic to validate
+-- @return topic, or false+error
+function mqtt.validate_publish_topic(t)
+	if type(t) ~= "string" then
```

Comment:
please follow such errors checking behavior: https://github.com/xHasKx/luamqtt/blob/master/mqtt/init.lua#L10

Reply:
same; result is input for assertions